[build-and-run] Migrate to Instruqt (#1)
DACH-NY/instruqt: topics/getting-started/02-your-first-feature/assignment.md

The return value of this component is the React `Form` element. This contains a dropdown menu to select a receiver from the following, a text field for the message content, and a Send button which triggers `submitMessage`.

There is again an important point here, in this case about how authorization is enforced. Due to the logic of the `SendMessage` choice, it is impossible to send a message to a user who is not following us (even if you could somehow access their `User` contract). The assertion that elem sender following in `SendMessage` ensures this: no mistake or malice by the UI programmer could breach this.

MainView Component
=====

Finally, we can see these components come together in the `MainView` component. We want to add a new panel to house our messaging UI. Open the `ui/src/components/MainView.tsx`{{open}} file and start by adding imports for the two new components.

> <pre class="file" data-target="clipboard">

Reviewer: ```suggestion
```javascript
```

import MessageEdit from './MessageEdit';
import MessageList from './MessageList';

> </pre>

Reviewer: ```suggestion
```
```

Next, find where the `Network Segment` closes, towards the end of the component. This is where we'll add a new `Segment` for `Messages`. Make sure you've saved the file after copying the code.

> <pre class="file" data-target="clipboard">

Reviewer: ```suggestion
```javascript
```

            &lt;Segment&gt;
              &lt;Header as='h2'&gt;
                &lt;Icon name='pencil square' /&gt;
                &lt;Header.Content&gt;
                  Messages
                  &lt;Header.Subheader&gt;Send a message to a follower&lt;/Header.Subheader&gt;
                &lt;/Header.Content&gt;
              &lt;/Header&gt;
              &lt;MessageEdit
                followers={followers.map(follower =&gt; follower.username)}
                partyToAlias={partyToAlias}
              /&gt;
              &lt;Divider /&gt;
              &lt;MessageList partyToAlias={partyToAlias}/&gt;

>             &lt;/Segment&gt;

Reviewer: Please replace here `&gt;` and `&lt;` with `>` and `<`

> </pre>

Reviewer: ```suggestion
```
```

You can see we simply follow the formatting of the previous panels and include the new messaging components: `MessageEdit` supplied with the usernames of all visible parties as props, and `MessageList` to display all messages.

That is all for the implementation! Let's give the new functionality a spin.

Running the New Feature
=====

It's time to run the app with the newly added feature!

As we're running the build process from scratch for the project, we need to install the project dependencies again. This wouldn't be necessary in case where project dependencies would have been previously installed.

Let's install the dependencies by running the below commands in the second terminal:

```bash
cd create-daml-app/ui
npm install
```

Now we're ready to start the UI by running:

```bash
npm start
```

As a reminder, this starts the web UI connected to the running Daml Sandbox and JSON API server. In case where we would have an already running UI this step wouldn't be necessary.

> Once the web UI has been compiled and started, you should see `Compiled successfully!` in your terminal. You can now [open the UI tab](https://[[HOST_SUBDOMAIN]]-3000-[[KATACODA_HOST]].environments.katacoda.com), where you should see the same login page as before

Reviewer: ```suggestion
Once the web UI has been compiled and started, you should see `Compiled successfully!` in your terminal. You can now open the **Daml UI tab**, where you should see the same login page as before
```
